### ISSUE TITLE
Regression tests don't pass with upcoming 9.5 release

### ISSUE THREAD

**Issue by anarazel** [citusdata/citus#705]
```
CREATE MATERIALIZED VIEW mv_articles_hash AS
        SELECT * FROM articles_hash WHERE author_id = 1 WITH NO DATA;
-DEBUG:  predicate pruning for shardId 840001
-DEBUG:  Creating router plan
-DEBUG:  Plan is router executable
-ERROR:  scan directions other than forward scans are unsupported
```

Due to a bugfix this now works. I'm kind of inclined to just remove that test, rather than duplicate the output for the two versions.

**Comment by anarazel** [comment]
@mtuncer ^

**Comment by ozgune** [comment]
I'm copy/pasting the following conversation from Slack. @anarazel / @mtuncer -- is the following transcript related to this issue?

(from https://citus-public.slack.com/archives/general/p1470934609000633)

--

I copied over some .a files from a postgres source build into Postgres.app lib/, and I can build citus now. and src/test/regress passes, except for ….
<pre>
*** /Users/rthomas/git/citus/src/test/regress/expected/multi_router_planner.out 2016-08-10 14:52:16.000000000 -0400
--- /Users/rthomas/git/citus/src/test/regress/results/multi_router_planner.out  2016-08-11 12:56:19.000000000 -0400
***************
*** 1540,1549 ****
  -- materialized views with (NO DATA) is still not supported
  CREATE MATERIALIZED VIEW mv_articles_hash AS
        SELECT * FROM articles_hash WHERE author_id = 1 WITH NO DATA;
! DEBUG:  predicate pruning for shardId 840001
! DEBUG:  Creating router plan
! DEBUG:  Plan is router executable
! ERROR:  scan directions other than forward scans are unsupported

  -- router planner/executor is disabled for task-tracker executor
  -- following query is router plannable, but router planner is disabled
--- 1540,1546 ----
  -- materialized views with (NO DATA) is still not supported
  CREATE MATERIALIZED VIEW mv_articles_hash AS
        SELECT * FROM articles_hash WHERE author_id = 1 WITH NO DATA;
! ERROR:  relation "mv_articles_hash" already exists

  -- router planner/executor is disabled for task-tracker executor
  -- following query is router plannable, but router planner is disabled
</pre>

This seems to have been an artifact of an earlier test failing (multi_table_ddl) and once I reran tests with that test fixed, this test passed fine

**Comment by mtuncer** [comment]
I am also for removing the test.  This change would most likely to go into 9.6 if it hasn't made it yet.
 9.5.4 change log says

> Do not run the planner on the query contained in CREATE MATERIALIZED VIEW or CREATE TABLE AS when WITH NO DATA is specified (Michael Paquier, Tom Lane)
> 
> This avoids some unnecessary failure conditions, for example if a stable function invoked by the materialized view depends on a table that doesn't exist yet.

**Comment by mtuncer** [comment]
Made the change and created a PR

**Comment by mtuncer** [comment]
Fix is merged to both release 5.2 and master.
